node-src/lib/getFileHashes.ts

```diff
@@ -0,0 +1,71 @@
+import { close, open, read } from 'fs';
+import pLimit from 'p-limit';
+import { join } from 'path';
+import xxHashWasm, { XXHash, XXHashAPI } from 'xxhash-wasm';
+
+const hashFile = (buffer: Buffer, path: string, xxhash: XXHashAPI): Promise<string> => {
+  const BUFFER_SIZE = buffer.length;
+
+  return new Promise((resolve, reject) => {
+    const done = (fd: number, getResult: () => bigint) => {
+      let result: bigint = undefined;
+      close(fd, (closeErr) => {
+        if (closeErr) reject(closeErr);
+        else resolve(result.toString(16).padStart(16, '0'));
+      });
+      result = getResult();
+    };
+
+    const readIncremental = (fd: number, hash: XXHash<bigint>) => {
+      read(fd, buffer, null, BUFFER_SIZE, -1, (readErr, bytesRead) => {
+        if (readErr) {
+          return close(fd, () => reject(readErr));
+        }
+        if (bytesRead === BUFFER_SIZE) {
+          hash.update(buffer);
+          readIncremental(fd, hash);
+        } else {
+          hash.update(buffer.subarray(0, bytesRead));
+          done(fd, () => hash.digest());
+        }
+      });
+    };
+
+    open(path, 'r', (openErr, fd) => {
+      if (openErr) {
+        return reject(openErr);
+      }
+      read(fd, buffer, null, BUFFER_SIZE, -1, (readErr, bytesRead) => {
+        if (readErr) {
+          return close(fd, () => reject(readErr));
+        }
+        if (bytesRead < BUFFER_SIZE) {
+          // Do a single hash if the whole file fits into the buffer.
+          done(fd, () => xxhash.h64Raw(buffer.subarray(0, bytesRead)));
+        } else {
+          // Otherwise use incremental hashing.
+          const hash = xxhash.create64();
+          hash.update(buffer);
+          readIncremental(fd, hash);
+        }
+      });
+    });
+  });
+};
+
+export const getFileHashes = async (files: string[], dir = process.cwd()) => {
+  const limit = pLimit(48);
```

Comment:
This is a cool lib, but it's unclear what it does and why we're using until you look it up. Can we alias it or add a comment that gives more info. Also, what's the reason behind the choice of 48 as the concurrency limit?

Reply:
This is an alternative to the resource pool used in the [original implementation](https://github.com/tripadvisor/npm-packages/blob/main/packages/fast-hash-file/throttled.ts#L9-L12). However that library is hardly used/maintained so I swapped it out for [p-limit](https://www.npmjs.com/package/p-limit) which I've used before and is very popular and well maintained. I kept the 48 concurrency limit for no particular reason other than that I have no idea what would be a better value. We could make it an environment variable I suppose.